### PR TITLE
feat: Emulation System Input Validation

### PR DESCRIPTION
**Pull request by DerekMaggio**
# Overview

Add parsing of input file representing emulated system with pydantic.

# Changelog

- Add hardware model class structure with pydantic
  - HardwareModel
    - RobotModel
      - OT2
      - OT3 (Not yet implemented)
      - OT3Breakout (Not yet implemented)
    - ModuleModel
      - ThermocyclerModule
      - HeaterShakerModule
      - TemperatureModule
      - MagneticModule
- Add HardwareSpecifcAttributes class that contains attributes specific to each piece of individual hardware
- Add unittesting for all bottom level Models
- Add SystemConfigurationModel to represent an entire system

# Review requests

- Feedback on the architecture of the HardwareModel would be helpful

# Risk assessment

None, it doesn't actually do anything yet

### CODE REVIEW

**Comment by DerekMaggio** [comment]
- Still need to add tests for any validation on HardwareModels 
- Still need to add tests for SystemConfigurationModel

**Comment by DerekMaggio** [comment]
Example config file
```json
{
    "robot": {
        "ot-2": {
            "hardware": "ot2",
            "emulation-level": "driver",
            "source-type": "local",
            "source-location": "/home/derek-maggio/Documents/repos/opentrons",
            "hardware-specific-attributes": {}
        }
    },
    "modules": {
        "heater-shaker-1": {
            "hardware": "heater-shaker-module",
            "emulation-level": "firmware",
            "source-type": "remote",
            "source-location": "latest",
            "hardware-specific-attributes": {
                "mode": "stdin"
            }
        },
        "thermocycler-1": {
            "hardware": "thermocycler-module",
            "emulation-level": "driver",
            "source-type": "local",
            "source-location": "/home/derek-maggio/Documents/repos/opentrons",
            "hardware-specific-attributes": {
                "lid-temperature": {
                    "degrees-per-tick": 3.0
                }
            }
        },
        "temp-1": {
            "hardware": "temperature-module",
            "emulation-level": "driver",
            "source-type": "local",
            "source-location": "/home/derek-maggio/Documents/repos/opentrons",
            "hardware-specific-attributes": {
                "temperature": {
                    "degrees-per-tick": 5.0
                }
            }
        },
        "mag-1": {
            "hardware": "magnetic-module",
            "emulation-level": "driver",
            "source-type": "local",
            "source-location": "/home/derek-maggio/Documents/repos/opentrons",
            "hardware-specific-attributes": {}
        }
    }
}
```

**Review by amitlissack — COMMENTED** [review]
Looks really solid. Just left a note about emulation level.

File: emulation_system/emulation_system/compose_file_creator/input/settings.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    THERMOCYCLER = "thermocycler-module"
+    TEMPERATURE = "temperature-module"
+    MAGNETEIC = "magnetic-module"
+    OT2 = "ot2"
+    OT3 = "ot3"
+
+
+class EmulationLevel(str, Enum):
+    """The emulation level of the emulator"""
+    FIRMWARE = "firmware"
```
[the remark this answers is not in the document]
Comment (DerekMaggio) [reply emulation_system/emulation_system/compose_file_creator/input/settings.py:19]: Oh yeah! Addressing now

[the remark this answers is not in the document]
Comment (DerekMaggio) [reply emulation_system/emulation_system/compose_file_creator/input/settings.py]: https://github.com/Opentrons/opentrons-emulation/pull/33/commits/44397ea4a8ac5c24cb1a018e87c56963fbf34aaf

**Review by amitlissack — APPROVED** [review]
Nice work!

**Pull request merged**